StandaloneMmPkg/Drivers/StandaloneMmIplPei/StandaloneMmIplPei.c

```diff
@@ -137,20 +144,52 @@ Communicate (
   IN OUT UINTN                           *CommSize
   )
 {
-  EFI_STATUS           Status;
-  PEI_SMM_CONTROL_PPI  *SmmControl;
-  UINT8                SmiCommand;
-  UINTN                Size;
+  EFI_STATUS                  Status;
+  PEI_SMM_CONTROL_PPI         *SmmControl;
+  UINT8                       SmiCommand;
+  UINTN                       Size;
+  EFI_SMM_COMMUNICATE_HEADER  *CommunicateHeader;
+  UINTN                       TempCommSize;
 
   DEBUG ((DEBUG_INFO, "StandaloneSmmIpl Communicate Enter\n"));
 
   SmiCommand = 0;
   Size       = sizeof (SmiCommand);
 
+  //
+  // Check parameters
+  //
   if (CommBuffer == NULL) {
     return EFI_INVALID_PARAMETER;
   }
 
+  CommunicateHeader = (EFI_SMM_COMMUNICATE_HEADER *)CommBuffer;
+
+  if (CommSize == NULL) {
+    TempCommSize = OFFSET_OF (EFI_SMM_COMMUNICATE_HEADER, Data) + CommunicateHeader->MessageLength;

```

Comment:
Why set TempCommSize to max size?

Reply:
StandaloneMM core always used max size to copy data from FixedCommbuffer to Internal Comm buffer